docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,317 @@
+# ImageSegmentation.jl
+
+## Introduction
+
+Image Segmentation is the process of partitioning the image into regions that have similar attributes. Image segmentation has various applications e.g, medical image segmentation, image compression and is used as a preprocessing step in higher level vision tasks like object detection and optical flow. This package is a collection of image segmentation algorithms written in Julia.
+
+## Installation
+
+```julia
+Pkg.add("ImageSegmentation")
+```
+
+## Example
+
+Let's see an example on how to use the segmentation algorithms in this package. We will try to seperate the deer, the ground and the sky in the image below using seeded region growing algorithm. This algorithm initializes the regions as the user-provided seed points and iteratively adds pixels to each region.
+
+![Original](assets/segmentation/deer.jpg) 
+
+```julia
+using Images, ImageSegmentation, ImageView
+
+img = load("deer.jpg")
+seeds = [(CartesianIndex(75, 250), 1), (CartesianIndex(150,165), 2), (CartesianIndex(200, 300), 3)]
+segments = seeded_region_growing(img, seeds)
+imshow(map(i->segments.segment_means[i], segments.image_indexmap))
+```
+
+![Original](assets/segmentation/deer_seg1.jpg) 
+
+All the segmentation algorithm (except Fuzzy C-means) return a struct `SegmentedImage` that stores the segmentation result. `SegmentedImage` contains a list of applied labels and an array containing the assigned label for each pixel. `seeded_region_growing` needs the initial seed points from the user. Some algorithms only need to know the number of segments or level of detail in segmentation. For example, `felzenszwalb` only needs a single parameter k which controls the size of segments. Larger k will result in bigger segments.
+
+```julia
+using Images, ImageSegmentation, ImageView
+
+img = load("deer.jpg")
+segments = felzenszwalb(img, 100)
```

Comment:
```min_size``` is the minimum segment size in pixels. 

From the paper
```
 In practice k sets a scale of observation, in that a larger k causes a preference
 for larger components. Note, however, that k is not a minimum component size. 
Smaller components are allowed when there is a sufficiently large difference between
 neighboring components.
```
I don't think ```k``` has any unit that is physically meaningful. I agree that we should add some information about reasonable values of ```k```.